requests_mv_integrations/support/tune_request.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#  @copyright 2016 TUNE, Inc. (http://www.tune.com)
+#  @namespace requests_mv_integrations
+
+from logging import getLogger
+import grequests
+import requests
+from requests.adapters import (HTTPAdapter, DEFAULT_POOLSIZE)
+from requests.packages.urllib3.util.retry import Retry
+from requests_mv_integrations.support import (
+    REQUEST_RETRY_HTTP_STATUS_CODES
+)
+
+log = getLogger(__name__)
+
+
+class TuneRequest(object):
+    POOL_SIZE = DEFAULT_POOLSIZE
+    session = None
+    request_buffer = []
+
+    def __init__(
+        self,
+        retry_tries=3,
+        retry_backoff=0.1,
+        retry_codes=None
+    ):
+        self.session = requests.session()
+
+        if retry_codes is None:
+            retry_codes = set(REQUEST_RETRY_HTTP_STATUS_CODES)
+
+        self.session.mount(
+            'http',
+            HTTPAdapter(
+                max_retries=Retry(
+                    total=retry_tries,
+                    backoff_factor=retry_backoff,
+                    status_forcelist=retry_codes,
+                ),
+            ),
+        )
+
+    def request(
+        self,
+        request_method,
+        request_url,
+        **kwargs
+    ):
+        response = self.session.request(
+            method=request_method,
+            url=request_url,
+            **kwargs
+        )
+
+        return response
+
+    def request_safe(
```

Comment:
per our discussion this method should just be removed infavor of the non-safe version